### ISSUE TITLE
Issue with 0.2.1 Program type already present: okhttp3.Address

### ISSUE THREAD

**Issue by WallBB** [stellar/java-stellar-sdk#111]
Hello,
I am getting following error with the 0.2.1 version of the java SDK

```
Program type already present: okhttp3.Address
Message{kind=ERROR, text=Program type already present: okhttp3.Address, sources=[Unknown source file], tool name=Optional.of(D8)}
```
My build.gradle is as follows
```
apply plugin: 'com.android.application'

android {
    compileSdkVersion 28
    defaultConfig {
        applicationId "com.example.dep.appil"
        minSdkVersion 21
        targetSdkVersion 28
        versionCode 1
        versionName "1.0"
        multiDexEnabled true
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'com.android.support:appcompat-v7:28.0.0-alpha3'
    implementation 'com.android.support:support-v4:28.0.0-alpha3'
    implementation 'com.android.support:design:28.0.0-alpha3'
    implementation 'com.android.support.constraint:constraint-layout:1.1.2'
    implementation 'com.github.stellar:java-stellar-sdk:0.2.1'

    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.2'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
}
```
Can you help me with the issue ?
I am not sure what is causing this issue

**Comment by bartekn** [comment]
Does it solve your issue: https://stackoverflow.com/a/50003848/9812359 ?

**Comment by WallBB** [comment]
Hi batekn,
Once I add that, I get issue related to unable to merge dex files and for that I have addedmultiDexEnabled true but it didn't fix the issue and I usually like project to be clean without Warnings.and 
I wanted to use this API in my existing application and after spending 2 days with this, I created a new empty project with basic drawer activity and tested that it was compiling. I then added the repository and implementation and started getting above issues in that case as well. Once I commented above API, my empty project build didn't had any problem.

The above gradle file is empty project with basic drawer activity. I tried it with a blank page activity as well and it didn't work.

**Comment by FunshoFab** [comment]
I also have this issue. @WallBB Have you found a fix yet?

**Comment by AltcoinN** [comment]
@FunshoFab The issue is still there when using jitpack and my suggestion is that author needs to fix that.
What I did is use the jar instead of the jitpack and it worked fine :)
https://github.com/stellar/java-stellar-sdk/releases

**Comment by WallBB** [comment]
@FunshoFab The issue is still there when using jitpack and my suggestion is that author needs to fix that.
What I did is use the jar instead of the jitpack and it worked fine.
https://github.com/stellar/java-stellar-sdk/releases

**Comment by jillesvangurp** [comment]
This should be addressed by #123 which shades the dependencies.

**Comment by dabitdev** [comment]
@jillesvangurp will the `shading the dependencies` make the AAR not fat anymore?

I just opened similar issue yesterday #156

**Comment by jillesvangurp** [comment]
We're shading and relocating the jars to a different package, so yes it makes the jar fat and it would break if you remove that because it depends on the moved packages. This is intentional because there have been some classpath conflicts with common stuff in combination with e.g. spring boot. This allows that to co-exist with other stuff on the classpath.

My latest pr #147 , which was just merged removes a lot of those problematic dependencies. Jersey was pulling in a lot of crap transitively. So, the new jar should be a lot more reasonable. Still some stuff that might conflict for some people. But jersey was plain nasty stuff.

But we could consider removing the shading stuff probably @bartekn, what do you think?

**Comment by dabitdev** [comment]
@WallBB to make it work you have to disable the transitive dependencies because stellar sdk is clashing with itself, check #156 

```
implementation("com.github.stellar:java-stellar-sdk:$stellarVersion") {
        transitive = false
    }
```

**Comment by bartekn** [comment]
Should be fixed in [0.4.0](https://github.com/stellar/java-stellar-sdk/releases/tag/0.4.0).
